internal/app/init/pkg/ntpd/ntpd.go

```diff
@@ -0,0 +1,93 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package ntpd
+
+import (
+	"log"
+	"math/rand"
+	"syscall"
+	"time"
+
+	"github.com/autonomy/talos/internal/pkg/userdata"
+	"github.com/beevik/ntp"
+)
+
+// https://access.redhat.com/solutions/39194
+// Using the above as reference for setting min/max
+const (
+	MAXPOLL = 1000
+	MINPOLL = 20
+	// TODO: Once we get naming sorted we need to apply
+	// for a project specific address
+	// https://manage.ntppool.org/manage/vendor
+	DEFAULTSERVER = "pool.ntp.org"
+)
+
+// New instantiates a new ntp instance against a given server
+// If no servers are specified, the default will be used
+func New(data *userdata.NTPd) *NTP {
+	server := DEFAULTSERVER
+
+	// Check if ntp servers are defined
+	if data != nil && data.Server != "" {
+		server = data.Server
+	}
+
+	log.Printf("setting ntp server to %s", server)
+	return &NTP{Server: server}
+}
+
+// NTP contains a server address
+// and the most recent response from a query
+type NTP struct {
+	Server   string
+	Response *ntp.Response
+}
+
+// Daemon runs the control loop for query and set time
+// We dont ever want the daemon to stop, so we only log
+// errors
+func (n *NTP) Daemon() {
+	var err error
+	rando := rand.New(rand.NewSource(time.Now().UnixNano()))
+	ticker := time.NewTicker(time.Duration(rando.Intn(MAXPOLL)+MINPOLL) * time.Second)
+
+	for {
+		select {
+		case <-ticker.C:
+			// Set some variance with how frequently we poll ntp servers
+			if err = n.Query(); err != nil {
```

Comment:
I think I would prefer `Query` to return a response and then `SetTime` to accept that as a param, but I can live with this. I feel like there is chance for stale data in the struct. Granted the way we are invoking it here prevents that.